spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/rnn/Attention.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn.rnn
+
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.nn.{Sequential, keras, _}
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity, TensorModule}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{Shape, T}
+
+import scala.reflect.ClassTag
+
+/**
+ * Implementation of multiheaded attention and self-attention layers.
+ * @param hidden_size hidden size
+ * @param num_heads heads number
+ * @param attention_dropout
+ */
+private[nn] class AttentionLayer[T: ClassTag](
+  hidden_size: Int, num_heads: Int, attention_dropout: Float)
+ (implicit ev: TensorNumeric[T]) extends BaseModule[T] {
+
+  override var model : Module[T] = buildModel()
+
+  private def buildModel(): Module[T] = {
+    // InputX with shape [batch_size, length_x, hidden_size].
+    // InputY with shape [batch_size, length_x, hidden_size]
+    // for self attention, InputX and InputY should be same.
+    // Bias is attention bias that will be added to the result of the dot product.
+    val inputX = Input()
+    val inputY = Input()
+    val inputBias = Input()
+
+    val q_dense_layer = TransformerOperation.dense(
```

Comment:
we should use java naming convention and void c/c++ one.

Reply:
ok, changed